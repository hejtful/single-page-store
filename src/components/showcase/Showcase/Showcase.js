import React from 'react';

import { ShowcaseItem } from '../ShowcaseItem';

import styles from './Showcase.module.css';

export const Showcase = ({ items, onAddItemToCart }) => (
  <div className={styles.showcase}>
    {items?.length ? (
      items.map((item) => (
        <ShowcaseItem {...item} key={item.id} onAdd={onAddItemToCart} />
      ))
    ) : (
      <div className={styles.emptyWrapper}>
        No products are currently available.
      </div>
    )}
  </div>
);
