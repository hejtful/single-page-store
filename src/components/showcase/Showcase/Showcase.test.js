import React from 'react';
import { render } from '@testing-library/react';

import { Showcase } from './Showcase';

const initialShowcaseItem = {
  id: '1',
  title: 'Test product title',
  price: 0.25,
  image: 'https://test.image.jpg',
  description: 'Test product description quite long',
};

const additionalShowcaseItem = {
  id: '2',
  title: 'Additional item title',
  price: 1000,
  image: 'https://additional.image.jpg',
  description: 'Shorter description',
};

const methods = {
  onAddItemToCart: jest.fn(),
};

it('renders a message indicating no products are available if items prop is undefined', () => {
  const props = {};

  const { container } = render(<Showcase {...props} {...methods} />);

  expect(container).toMatchSnapshot();
});

it('renders a message indicating no products are available if items prop is an empty array', () => {
  const props = {
    items: [],
  };

  const { container } = render(<Showcase {...props} {...methods} />);

  expect(container).toMatchSnapshot();
});

it('renders a single product', () => {
  const props = {
    items: [initialShowcaseItem],
  };

  const { container } = render(<Showcase {...props} {...methods} />);

  expect(container).toMatchSnapshot();
});

it('renders multiple products', () => {
  const props = {
    items: [initialShowcaseItem, additionalShowcaseItem],
  };

  const { container } = render(<Showcase {...props} {...methods} />);

  expect(container).toMatchSnapshot();
});
