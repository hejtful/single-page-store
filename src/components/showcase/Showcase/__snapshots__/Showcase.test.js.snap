// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a message indicating no products are available if items prop is an empty array 1`] = `
<div>
  <div
    class="showcase"
  >
    <div
      class="emptyWrapper"
    >
      No products are currently available.
    </div>
  </div>
</div>
`;

exports[`renders a message indicating no products are available if items prop is undefined 1`] = `
<div>
  <div
    class="showcase"
  >
    <div
      class="emptyWrapper"
    >
      No products are currently available.
    </div>
  </div>
</div>
`;

exports[`renders a single product 1`] = `
<div>
  <div
    class="showcase"
  >
    <div
      class="wrapper"
    >
      <div
        class="imageWrapper"
      >
        <img
          alt="Test product title product"
          class="image"
          src="https://test.image.jpg"
        />
      </div>
      <div
        class="infoWrapper"
      >
        <div
          class="titleWrapper"
        >
          <h4
            class="title"
          >
            Test product title
          </h4>
          <span>
            0.25
            $
          </span>
        </div>
        <p
          class="description"
        >
          Test product description quite long
        </p>
        <button
          type="button"
        >
          Add to Order
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders multiple products 1`] = `
<div>
  <div
    class="showcase"
  >
    <div
      class="wrapper"
    >
      <div
        class="imageWrapper"
      >
        <img
          alt="Test product title product"
          class="image"
          src="https://test.image.jpg"
        />
      </div>
      <div
        class="infoWrapper"
      >
        <div
          class="titleWrapper"
        >
          <h4
            class="title"
          >
            Test product title
          </h4>
          <span>
            0.25
            $
          </span>
        </div>
        <p
          class="description"
        >
          Test product description quite long
        </p>
        <button
          type="button"
        >
          Add to Order
        </button>
      </div>
    </div>
    <div
      class="wrapper"
    >
      <div
        class="imageWrapper"
      >
        <img
          alt="Additional item title product"
          class="image"
          src="https://additional.image.jpg"
        />
      </div>
      <div
        class="infoWrapper"
      >
        <div
          class="titleWrapper"
        >
          <h4
            class="title"
          >
            Additional item title
          </h4>
          <span>
            1000
            $
          </span>
        </div>
        <p
          class="description"
        >
          Shorter description
        </p>
        <button
          type="button"
        >
          Add to Order
        </button>
      </div>
    </div>
  </div>
</div>
`;
